kmcd-props.md

```diff
@@ -326,6 +354,50 @@ The property checks if a successful `Pot . join` is preceded by a `TimeStep` mor
     rule derive(zeroTimePotInterestEnd, LogNote( _ , Pot . drip   )) => zeroTimePotInterest
 ```
 
+### Flap dai consistency
+
+```k
+    syntax ViolationFSM ::= "flapDaiConsistency"
+ // --------------------------------------------
+    rule derive(flapDaiConsistency, FlapKick(_, _, _, _)) => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapTend(_, _, _))    => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapDeal(_))          => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
+    rule derive(flapDaiConsistency, FlapYank(_))          => Violated(flapDaiConsistency) requires notBool(flapDaiGtOrEtSumOfFlapLots())
```

Comment:
Instead of matching on all these events, you could just match on the `Measure(_)` event. That way it gets checked every step and not only when `Flap` steps are taken. This is good for 2 reasons:

-   Let's you have a single rule instead of 4.
-   You can then add the given measures you want (`sumOfAllFlapBids` and `sumOfAllFlapLots`) to the `Measure` record, so that this is not being turned into a contextual function.

Reply:
Done.